### PR TITLE
First shot at read api.

### PR DESCRIPTION
**Pull request by aenario**
So this is the first shot at GetDoc API

It is based on 3 commits by @nono which introduce the instance concept.

Some design decisions : 
- a couchdb packages at cozy-stack/couchdb
- Introducing the notion of database prefix in instance, which joined with doctype give use the database name
- using `gin.ErrorLogger` to output json errors
- Changed status test for healthy and added couchdb to travis

Left TODO :
- the database naming should be better analysed, for now it will fail if there is a character outside of `_$()+-/a-zA-Z0-9` in host or doctype, or if the host doesnt start with a `a-z`.
- Need more code comment, will be working on it in parallel with review

@nono, there is a merge conflict with master. I could fix-it by merging master into this branch, or rebasing this branch on master, but it gets messy with your commits included here.

### CODE REVIEW

File: couchdb/couchdb.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil {
+		return &CouchdbError{http.StatusInternalServerError,
+      []byte("{\"error\":\"Wrong configuration for couchdbserver\"}")};
+	}
+	resp, err := CouchdbClient.Do(req)
+	if err != nil {
+		return &CouchdbError{http.StatusServiceUnavailable,
+      []byte("{\"error\":\"No couch to seat on.\"}")}
+	}
+  body, err := ioutil.ReadAll(resp.Body)
```
Comment (nono) [inline couchdb/couchdb.go]: I think you need to close the body:

```go
defer resp.Body.Close()
```

**Comment by nono** [comment]
I think you should merge master in this branch to resolve the conflict

File: couchdb/couchdb.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+func DocURL(dbprefix, doctype, id string) string{
+  return makeDBURL(dbprefix, doctype) + "/" + makeDocID(doctype, id)
+}
+
+func GetDoc(dbprefix, doctype, id string, out interface{}) error {
+  url := DocURL(dbprefix, doctype, id)
+  fmt.Printf("[couchdb request] %v\n", url)
+	req, err := http.NewRequest("GET", url, nil)
+  req.Header.Add("Accept", "application/json")
```
Comment (jinroh) [inline couchdb/couchdb.go]: this `req.Header.Add` should probably be after the `if err != nil`.

File: couchdb/couchdb.go (hunk 1 above)
Comment (aenario) [inline couchdb/couchdb.go]: done in https://github.com/cozy/cozy-stack/pull/22/commits/91542eb03e8eda871b47723dda22bd71b55880b7

File: couchdb/couchdb.go (hunk 2 above)
Comment (aenario) [inline couchdb/couchdb.go]: done in https://github.com/cozy/cozy-stack/pull/22/commits/91542eb03e8eda871b47723dda22bd71b55880b7

**Comment by aenario** [comment]
OK, I figured out what was wrong with my setup (wrong GOPATH in profile made the linters atom plugins fail silently). So now, gofmt get applied, gometalinter too and travis is happy, which makes me happy.

Both your reviews comment have been fixed in https://github.com/cozy/cozy-stack/pull/22/commits/f214b15ddbd8bc0bb24a11c579483c30032820ee

**Pull request merged**